src/Lorentz/Contracts/RegistryDAO.hs

```diff
@@ -139,6 +145,45 @@ decisionLambda = do
         swap; toField #cpMaxProposalSize; ifSome (setField #ceMaxProposalSize) nop
         stSetField #sExtra
         nil
+    , #cUpdateReceiversType /-> do
+        stackType @(UpdateReceivers ': (DAO.ProposalKey (RegistryDaoProposalMetadata k v)) : store : _)
+        dip drop
+        stackType @(UpdateReceivers ': store : _)
+
+        -- Push a boolean value to indicate if we are adding addreses
+        -- or removing them. True to add and False to receive.
+        caseT
+          ( #cAddReceivers /-> push True
+          , #cRemoveReceivers /-> push False
+          )
+        stackType @(Bool ': [Address] ': store : _)
+        duupX @3; stToField #sExtra
+
+        stackType @(RegistryDaoContractExtra k v ': Bool ': [Address] ': store : _)
+        dup
+        toField #ceProposalReceivers
+        stackType @(Set Address ': RegistryDaoContractExtra k v ': Bool ': [Address] ': store : _)
+        dip swap
+        duupX @4;
+
+        -- Now we have the input addresses at the top
+        -- The current set of addresse below it,
+        -- and the boolean that indicate if we are adding or removing, under it.
+        stackType @([Address] : Set Address ': Bool ': RegistryDaoContractExtra k v ': [Address] ': store : _)
+        iter $ do
+          -- We fetch the boolean operation indicator from the stack
+          dip $ duupX @2
+          update
+
+        -- Now we have the updated set at the top. We update the storage with it.
+        stackType @(Set Address ': Bool ': RegistryDaoContractExtra k v ': [Address] ': store : _)
+        dip drop
+        stackType @(Set Address ': RegistryDaoContractExtra k v ': [Address] ': store : _)
+        setField #ceProposalReceivers
+        stackType @(RegistryDaoContractExtra k v ': [Address] ': store : _)
+        dip drop
+        stSetField #sExtra
```

Comment:
```suggestion
        dip (drop # stGetField #sExtra # getField #ceProposalReceivers)
        stackType @(UpdateReceivers ': Set Address ': RegistryDaoContractExtra k v ': store : _)

        -- Push a boolean value to indicate if we are adding addresses
        -- or removing them. True to add and False to receive.
        caseT
          ( #cAddReceivers /-> push True
          , #cRemoveReceivers /-> push False
          )
        swap
        stackType @([Address] ': Bool ': Set Address ': RegistryDaoContractExtra k v ': store : _)

        -- Now we have the input addresses at the top
        -- The boolean that indicate if we are adding or removing below it
        -- and the current set of addresse, under it.
        iter $ do
          -- We put a copy of the boolean operation indicator on top of the stack
          duupX @2
          dip update
        drop

        stackType @(Set Address ': RegistryDaoContractExtra k v ': store : _)
        setField #ceProposalReceivers
        stSetField #sExtra
```
I'm not sure how much of this would be optimized away by the optimizer anyway, but since you are so close this rearranging may be enough.

Reply:
Had to optimized a couple of more bytes, but it appear to be withing the limits now. Tests passed now.